flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNITCase.java

```diff
@@ -79,20 +86,45 @@ public void testPerJobMode() throws Exception {
 
 			env.addSource(new InfiniteSource())
 				.shuffle()
-				.addSink(new DiscardingSink<Integer>());
+				.addSink(new DiscardingSink<>());
 
 			final JobGraph jobGraph = env.getStreamGraph().getJobGraph();
 
 			File testingJar = YarnTestBase.findFile("..", new TestingYarnClusterDescriptor.TestJarFinder("flink-yarn-tests"));
 
 			jobGraph.addJar(new org.apache.flink.core.fs.Path(testingJar.toURI()));
 
-			ClusterClient<ApplicationId> clusterClient = yarnClusterDescriptor.deployJobCluster(
-				clusterSpecification,
-				jobGraph,
-				true);
+			ApplicationId applicationId = null;
+			ClusterClient<ApplicationId> clusterClient = null;
+
+			try {
+				clusterClient = yarnClusterDescriptor.deployJobCluster(
+					clusterSpecification,
+					jobGraph,
+					false);
+				applicationId = clusterClient.getClusterId();
+
+				assertThat(clusterClient, is(instanceOf(RestClusterClient.class)));
+				final RestClusterClient<ApplicationId> restClusterClient = (RestClusterClient<ApplicationId>) clusterClient;
+
+				final CompletableFuture<JobStatus> jobStatus = restClusterClient.getJobStatus(jobGraph.getJobID());
+				assertThat(jobStatus.get(), is(JobStatus.RUNNING));
```

Comment:
Should we wait for `RUNNING`? or say would it be unstable when we request jobStatus before the job switch to RUNNING?

Reply:
In this case it works because we use the standalone leader election service which triggers as the first action the job scheduling (before we can request the job). However, I agree that a `NoDataSource` would be better for stability reasons. Will change it.